src/call-context.ts

```diff
@@ -88,34 +88,19 @@ export class CallContext {
     if (!this.#vars.has(name)) {
       return null;
     }
-    return this.read(this.#vars.get(name) as number);
+    return this.#vars.get(name) || null;
   }
 
   /**
    * Set a variable to a given string or byte array value. Returns the start
    * address of the variable. The start address is reused when changing the
    * value of an existing variable.
-   *
-   * @returns bigint
    */
-  setVariable(name: string, value: string | Uint8Array): bigint {
-    const newIdx = this[STORE](value);
-    if (newIdx === null) {
-      return 0n;
-    }
-
-    // Re-use the old address mapping.
-    const oldIdx = this.#vars.get(name) ?? null;
-    if (oldIdx !== null) {
-      this.#blocks[oldIdx] = this.#blocks[newIdx];
-      this.#blocks[newIdx] = null;
-      if (newIdx === this.#blocks.length - 1) {
-        this.#blocks.pop();
-      }
+  setVariable(name: string, value: string | Uint8Array) {
+    if (typeof value === 'string'){
+      value = this.#encoder.encode(value);
     }
-
-    this.#vars.set(name, oldIdx ?? newIdx);
-    return Block.indexToAddress(oldIdx ?? newIdx);
+    this.#vars.set(name, new PluginOutput(value.buffer));
```

Comment:
I think there might be something tricky to navigate around here with background workers. I _think_ this would fail in the following case:

```js
createPlugin('path/to/some/plugin', {
  runInWorker: true,
  functions: {
    'a namespace': {
       async hostfunc(context) {
         context.getVariable("foo") // should return BAZ but returns null
         context.setVariable("foo", "bar")
       }
    }
  }
})
```

Where the pseudocode for the plugin would be something like:

```js
Var.set("foo", "BAZ")
const {hostfunc} = Host.getFunctions();

func();
Var.get("foo") // should return "bar", but instead returns "BAZ"
```

because the worker and main thread only sync memory held in `#blocks`. I can see about writing this up as a test if that'd be helpful!

Reply:
Ah interesting, yeah I'm not 100% clear on some of the memory details - if you could add that test it would be super helpful!

In the meantime I will poke around and try to understand my changes a little better